packages/api/cms-api/src/content-scope/scope.guard.ts

```diff
@@ -61,7 +69,7 @@ export class ScopeGuard implements CanActivate {
                 if (subjectScope === undefined) throw new Error("Scope not found");
                 if (args.scope) {
                     // args.scope also exists, check if they match
-                    if (!isEqual(args.scope, subjectScope)) {
+                    if (!isEqual(args.scope, JSON.parse(JSON.stringify(subjectScope)))) {
```

Comment:
Actually I was looking for a better way, I couldn't find it. If the Entity uses an Embeddable (https://mikro-orm.io/docs/embeddables) for the scope-Field it just doesn't work without the above "fix".

Reply:
We could write our own compare:

```ts
if(!isEqual(Object.keys(args.scope).every(key => args.scope[key] === subjectScope[key]))) {
  ...
}
```

Reply:
Furthermore, the automatic check of the "scope"-field in the entity feels like a bit of magic. Maybe it just should work in Conjunction with the `ScopedEntity`-decorator (which then also can serialize the result correctly to avoid this JSON.parse/JSON.stringify-workaround).

Reply:
> Maybe it just should work in Conjunction with the ScopedEntity-decorator 

how do you imagine this?

Reply:
`@ScopedEntity` could have an optional `scopesEqual(requestScope, entityScope) => boolean` that defaults to lodash.isequal, for instance